Objects/genobject.c

```diff
@@ -718,15 +720,18 @@ gen_set_name(PyObject *self, PyObject *value, void *Py_UNUSED(ignored))
                         "__name__ must be set to a string object");
         return -1;
     }
-    Py_XSETREF(op->gi_name, Py_NewRef(value));
+    Py_BEGIN_CRITICAL_SECTION(self);
```

Comment:
Would you mind adding a comment explaining why XDecRefDelayed() is needed here? The name should be a simple Python str object, no?